ios/Classes/PluginConfig.swift

```diff
@@ -1,30 +1,174 @@
+/* Copyright Airship and Contributors */
 import Foundation
 import AirshipKit
 
+
+
 class PluginConfig {
     
-    private enum Keys : String {
+    private enum Keys: String {
         case appKey
         case appSecret
     }
     
     private static let defaults = UserDefaults(suiteName: "com.urbanairship.flutter")!
     
     static var appKey: String? {
-        get { return read(.appKey) }
-        set { write(.appKey, value: newValue) }
+        get {
+            return read(.appKey)
+        }
+        set {
+            write(.appKey, value: newValue)
+        }
     }
     
     static var appSecret: String? {
-        get { return read(.appSecret) }
-        set { write(.appSecret, value: newValue) }
+        get {
+            return read(.appSecret)
+        }
+        set {
+            write(.appSecret, value: newValue)
+        }
     }
     
     private static func read<T>(_ key: Keys) -> T? {
         return defaults.object(forKey: key.rawValue) as? T
     }
     
     private static func write(_ key: Keys, value: Any?) {
-        defaults.set(value, forKey: key.rawValue)
+        if let value = value {
+            defaults.set(value, forKey: key.rawValue)
+        } else {
+            defaults.removeObject(forKey: key.rawValue)
+        }
+    }
+}
+
+
+
+extension Config {
+    static func parse(_ configDict: AirshipConfig) throws -> Config {
+        let airshipConfig = Config.default()
+        
+        airshipConfig.parseDefaultEnv(configDict.defaultEnv)
```

Comment:
Not sure if Swift protobuf is similar to Kotlin in terms of empty/default values vs nullable fields, but do we need to handle those here, too?

Reply:
Applying all android changes into IOS